Module & Tests

alonweissfeld opened the pull request:
@kensaggy @avinoamr

Commit:
Package files

Commit:
Added error test

Commit:
Parsing test

Commit:
Updated README

kensaggy commented on test.js:
```diff
@@ -0,0 +1,78 @@
+ * @param {Mixed} input - given input
+ * @return {String} returns the quoted input
+ */
+function quotes ( input ) {
```
This doesn't quote a given output... this simply converts any input to a string. If the input is a string it doesn't add quotes.

kensaggy commented on test.js:
```diff
@@ -0,0 +1,78 @@
+ * @param {String} input - given input
+ * @return {Mixed} returns the parsed object
+ */
+function parse ( input ) {
```
Not sure why you need this method at all... simply to save you from adding `JSON.` before calling `parse()`?

kensaggy commented on test.js:
```diff
@@ -0,0 +1,78 @@
+    it( 'error', function ( done ) {
+        var badJsonString = '{"name":"somename}';
+
+        Promise.resolve( badJsonString )
```
Not sure why you need `Promise`/`bluebird` for code that is completely synchronous... seems like a big dependency to add just for calling `.then` (which isn't necessary for synch code)

alonweissfeld replied:
@kensaggy I think it's more readable and tidy, especially when handling errors that might occur during the test, but since it is a big dependency (and for the minimal use purposes in this test) I'm removing it.

kensaggy commented on test.js:
```diff
@@ -0,0 +1,78 @@
+var MIN_SAFE_INTEGER = Number.MIN_SAFE_INTEGER;
+
+describe( 'Parse Big', function () {
+    it( 'error', function ( done ) {
```
This would generate a test output such as:

```
Parse big error:
 AssertionError ....
```

A test name should describe the desired behaviour, just as an example

``` js
it('throws an error on unparseable strings', funciton ()...
```

This would generate `Parse Big throws an error on unparseable strings` which reads much more easily.
Same comment goes for `parse` test.

kensaggy commented on package.json:
```diff
@@ -0,0 +1,25 @@
+  "dependencies": {
+    "bluebird": "^3.4.1",
+    "json-bigint": "^0.2.0",
+    "mocha": "^3.0.0",
```
This should be in a `devDependencies` - as its only used for testing

kensaggy commented on test.js:
```diff
@@ -0,0 +1,78 @@
+require( 'sugar' );
```
Again seems like an unnecessary dependency simply for using `.assign` (and for something as simply as converting to string... not even "heavy" string formatting.

When doing auxiliary modules try to keep dependencies to a minimum so they're as thin and lightweight as possible. Its even worth it sometimes to implement your own stuff just to have less dependencies.

kensaggy replied:
Package `sugarjs` no longer exist as a dependencies so shouldn't be required :)

kensaggy commented on .gitignore:
```diff
@@ -0,0 +1,27 @@
+
+# Dependency directory
+# https://www.npmjs.org/doc/misc/npm-faq.html#should-i-check-my-node_modules-folder-into-git
+node_modules
```
This is probably the only line that's actually needed in this `.gitignore` file.
It doesn't generate logs, it doesn't have "Runtime data", it doesn't use "jscoverage" or "JSCover", or any of the other stuff.

kensaggy commented:
Done reviewing.

kensaggy commented:
Also, not a big fan of the name `parse-big` - its so generic you can hardly tell what its doing.
a) what does it "parse"? what does "big" have to do with it.
b) it doesn't really supply any "parse" functionality - its more of an override (as soon as you require)

Commit:
Removed quotes method

Commit:
Removed parse method

Commit:
Removed Bluebird

Commit:
Better test descriptions

Commit:
Removed unnecessary dependencies

Commit:
Removed redundants from .gitignore

alonweissfeld commented:
@kensaggy Fixed all your comments and removed the redundant dependencies. 
Regarding the name, well it does functions as parsing big numbers... But I'll try to think of a better name. Also would be happy to hear suggestions, if you have something :)

Commit:
Renamed module

avinoamr commented on README.md:
````diff
@@ -1 +1,13 @@
+### Usage
+
+```javascript
+require( 'beyond-global-json' );
````
@alonweissfeld shouldn't it be:

``` javascript
require( 'beyond-global-json' ).overwrite()
```

?

alonweissfeld replied:
@avinoamr It's already invoking the `overwrite` method when being required:
`module.exports = overwrite()`

alonweissfeld commented:
@kensaggy I renamed the module to `beyond-global-json`. Bottom line, it overwrites the global `JSON` object - currently to help parsing big numbers, but other methods implementation could be added, or the helper module `json-bigint` could be changed. So giving it a name to associate _only_ with parsing big numbers is mistakable.

avinoamr commented:
Nothing much. I'm done.

kensaggy commented on test.js:
```diff
@@ -0,0 +1,54 @@
+require( 'sugar' );
+var assert = require( 'assert' );
+var Promise = require( 'bluebird' );
```
Package `bluebird` no longer exist as a dependencies so shouldn't be required :)

kensaggy commented:
Two comments regarding the test - you removed your dependency on `sugar` and `bluebird` but left the `require(...)` in the tests ;)

I'm not a big fan of the new name either (sorry!) I would have called it something like `panoply-json-override` but if @avinoamr didn't complain about the name.....its something I can live with.
Update: OK Truth be told I can't live with it :) Please change it for my sake... I know its a bit of work here and on the other PR as well. Thanks

Commit:
Removed required and unused dependencies

alonweissfeld commented:
@kensaggy Okay I admit the name is kinda stupid. Probably just felt creative at the moment. :) However I don't think it should have the `panoply` name in it since it doesn't help anyone with information and it's already sitting in `/panoplyio` so - `'/panoplyio/panoply-...` might be too much.
How about `global-json-override`?

Commit:
Better grammer: overwrites -> overrides

Commit:
Remoed trailing commas from package.json; new lines at EOF

Commit:
Change module's name

alonweissfeld commented:
@kensaggy Oops. Removed the required and unused dependencies. 
Also changed it to `global-json-override`. It the name is still not satisfying let me know

Commit:
Updated module name for README

kensaggy commented:
@alonweissfeld Good job :) Now you can publish the npm module
